libs/payments/stripe/src/lib/stripe.service.ts

```diff
@@ -3,10 +3,38 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import { Injectable } from '@nestjs/common';
+import {
+  SubscriptionCustomerIdDoesNotMatchCustomerIdError,
+  SubscriptionNotActiveError,
+} from './stripe.error';
+import { StripeManager } from './stripe.manager';
 
 @Injectable()
 export class StripeService {
-  constructor() {}
+  constructor(private stripeManager: StripeManager) {}
+
+  async applyPromoCodeToSubscription(
+    customerId: string,
+    subscriptionId: string,
+    promotionId: string
+  ) {
+    const subscription = await this.stripeManager.retrieveSubscription(
+      subscriptionId
+    );
+    if (subscription?.status !== 'active')
+      throw new SubscriptionNotActiveError();
+    if (subscription.customer !== customerId)
+      throw new SubscriptionCustomerIdDoesNotMatchCustomerIdError();
+
+    const updatedSubscription = await this.stripeManager.updateSubscription(
```

Comment:
What does this throw if the promo code is invalid, or if the promo code is restricted to a product that doesn't match the product on this subscription?

We should capture the Stripe error in both of these cases and re-throw our own library error in those cases.